sklearn/naive_bayes.py

```diff
@@ -379,8 +414,11 @@ def _partial_fit(self, X, y, classes=None, _refit=False,
             self.sigma_[i, :] = new_sigma
             self.class_count_[i] += N_i
 
+        class_prior = self.class_prior
+
         self.sigma_[:, :] += epsilon
-        self.class_prior_[:] = self.class_count_ / np.sum(self.class_count_)
+        self._update_class_prior(class_prior=class_prior)
```

Comment:
Maybe it would be more preferable to check this case in the `_update_class_prior` function itself.

We can either set `self.class_prior_` to None (or check if the variable exist) in the constructor and create an array of shape `(n_classes,)` in that case.

What do you think?